src/common/components/composite/forms/FormikUserSelector.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { FC, useState } from 'react';
+import { useField } from 'formik';
+import { FormHelperText, TextField } from '@mui/material';
+import Autocomplete, { autocompleteClasses } from '@mui/material/Autocomplete';
+import { useMessagingAvailableRecipientsLazyQuery } from '../../../../core/apollo/generated/apollo-hooks';
+import { User, UserFilterInput } from '../../../../core/apollo/generated/graphql-schema';
+import { UserSelectorView } from './FormikUserSelector/UserSelectorView';
+import { remove } from 'lodash';
+import SearchIcon from '@mui/icons-material/Search';
+import { gutters } from '../../../../core/ui/grid/utils';
+import GridContainer from '../../../../core/ui/grid/GridContainer';
+import useCurrentBreakpoint from '../../../../core/ui/utils/useCurrentBreakpoint';
+import GridProvider from '../../../../core/ui/grid/GridProvider';
+import { UserSelectedView } from './FormikUserSelector/UserSelectedView';
+
+const MAX_USERS_SHOWN = 10;
+const GRID_COLUMNS_DESKTOP = 6;
+const GRID_COLUMNS_MOBILE = 3;
+const uniqueId = () => Math.random().toString(36).slice(2);
+
+interface FormikUserSelectorProps {
+  name: string;
+  required?: boolean;
+  readonly?: boolean;
+  onChange?: (currentValue: string[]) => void;
+}
+
+export const FormikUserSelector: FC<FormikUserSelectorProps> = ({
+  name,
+  required,
+  readonly,
+  onChange,
+  ...containerProps
+}) => {
+  // This field is the array of user Ids
+  const [field, meta, helpers] = useField<string[] | undefined>(name);
+  const validationError = Boolean(meta.error) && meta.touched;
+
+  const [filter, setFilter] = useState<UserFilterInput>();
+
+  const [loadUsers, { data }] = useMessagingAvailableRecipientsLazyQuery({
+    variables: { filter, first: MAX_USERS_SHOWN },
+  });
+
+  const users = data?.usersPaginated.users ?? [];
+  // Filter out users that are already selected
+  const listedUsers = users.filter(user =>
+    field.value && field.value.length ? field.value.indexOf(user.id) === -1 : true
+  );
+  // Clear Autocomplete when a user is selected
+  const [autocompleteValue, setAutocompleteValue] = useState<User | null>(null);
+  // Autocomplete is re-rendered when we change the key property
+  // Setting name of the text input to a random string also disables autocomplete browser features
+  const [autocompleteKey, setAutocompleteKey] = useState<string>(uniqueId());
+
+  const breakpoint = useCurrentBreakpoint();
+
+  const handleSelect = (user: Pick<User, 'id'> | null) => {
+    helpers.setTouched(true);
+
+    if (user === null) return;
+    let value = field.value;
+    if (!value || !Array.isArray(value)) {
+      value = [];
+    }
+    value.push(user.id);
+    helpers.setValue(value);
+    // Clear autocomplete on every select
+    setAutocompleteValue(null);
+    setAutocompleteKey(uniqueId()); // Force re-render
```

Comment:
I've had to add back the some random string generation for the name property of the TextField because Chrome insists in ignoring any attempt to disable autofill. but I think is cleaner thank what I had here